persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java

```diff
@@ -56,60 +50,56 @@ public class BalProjectUtils {
 
     private BalProjectUtils() {}
 
-    public static DiagnosticResult hasSemanticDiagnostics(Path projectPath) {
-        Package currentPackage;
-        BuildProject buildProject;
-        buildProject = BuildProject.load(projectPath.toAbsolutePath());
-        currentPackage = buildProject.currentPackage();
-        PackageCompilation compilation = currentPackage.getCompilation();
-        return compilation.diagnosticResult();
-    }
-
     public static EntityMetaData getEntitiesInBalFiles(String sourcePath) throws BalException {
         ArrayList<Entity> returnMetaData = new ArrayList<>();
         ArrayList<ModuleMemberDeclarationNode> returnModuleMembers = new ArrayList<>();
         Path dirPath = Paths.get(sourcePath);
-        List<Path> fileList;
-        Path clientEntitiesPath = Paths.get(sourcePath, KEYWORD_MODULES, KEYWORD_CLIENTS,
-                PATH_ENTITIES_FILE).toAbsolutePath();
-        File entitiesBal = new File(clientEntitiesPath.toString());
         try {
-            if (!entitiesBal.exists()) {
-                DiagnosticResult diagnosticResult = hasSemanticDiagnostics(dirPath);
-                if (diagnosticResult.hasErrors()) {
-                    StringBuilder errorMessage = new StringBuilder();
-                    errorMessage.append("Error occurred when validating the project. ");
-                    for (Diagnostic d : diagnosticResult.errors()) {
-                        errorMessage.append(System.lineSeparator());
-                        errorMessage.append(d.toString());
+            BuildProject buildProject = BuildProject.load(dirPath.toAbsolutePath());;
+            Package currentPackage = buildProject.currentPackage();
+            PackageCompilation compilation = currentPackage.getCompilation();
+            DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+            if (diagnosticResult.hasErrors()) {
+                StringBuilder errorMessage = new StringBuilder();
+                int count = 0;
+                errorMessage.append("Error occurred when validating the project. ");
+                for (Diagnostic d : diagnosticResult.errors()) {
+                    if (d.toString().contains("redeclared symbol")) {
```

Comment:
Shall we make sure that this diagnostic location is inside clients directory. Also use diagnostic code. That way even if the error messages changed we need not update our logic.

Reply:
Removed as it is not required since client directory is ignored in the compiler check

Reply:
Where is is removed, AFAIK all the modules are loaded into the project

Reply:
https://github.com/ballerina-platform/module-ballerina-persist/blob/19cf1ab39e290e65fc69cc543590d3dfe3a85240/compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java#L120

Reply:
I am not sure if you understood the original scenario. What I wanted to see is,
1. User defines an entity
2. User calls bal generate
3. User uses the client to write code (read)
4. User changes the entity
5. User runs bal persist generate.

Only the above scenario should be allowed. Any other compiler errors should result in error.

Reply:
This scenario is allowed. As we don't  validate the clients sub-module it won't throw an error in `create` function in clients .Unless the user has unchanged record declarations in main.bal(for example user has removed `int itemId` from the entity but it is still there in a record in `main.bal`) there is no error thrown for this case. so it is not required to check/skip for this specific error in the compilation.

Reply:
+1 for adding diagnostic code, but checking whether diagnostic location is "Clients" won't be required as the all the diagnostics from client folder is currently ignored.

Reply:
Only persist module diagnostics are ignored. There can be other diagnostics if the user has changed this files. However, due to changes coming with "generated" folder feature we can ignore this for now.